uber/menu.py

```diff
@@ -0,0 +1,51 @@
+from uber.common import *
+
+
+class MenuItem:
+    access = None   # list of permission levels allowed to display this menu
+    href = None     # link to render
+    submenu = None  # submenu to show
+    name = None     # name of Menu item to show
+
+    def __init__(self, href=None, access=None, submenu=None, name=None):
+        if submenu:
+            self.submenu = listify(submenu)
+        else:
+            self.href = href
+
+        self.name = name
+        self.access = access
+
+    def __getitem__(self, key):
+        for sm in self.submenu:
+            if sm.name == key:
+                return sm
+
+    def append_menu_item(self, m):
+        # if we aren't a submenu, convert us to one now
+        if not self.submenu and self.href:
+            self.submenu = [MenuItem(name=self.name, href=self.href)]
+            self.href = None
+
+        self.submenu.append(m)
+
+
+c.MENU = MenuItem(name='Root', submenu=[
```

Comment:
Yea, I was torn about including it in config.py originally.

Actually, truth be told the only reason this menu code is involved with config.py is so that it's automatically exportable to templates (everything in Config is).  That was kind of a cheap hack.

I like the idea of the menu code living in its own file so that it's just easier to look at (config.py is already xbox huge).  And what I might end up doing later is having the menu export function be injected into the template data by adding it to the ```renderable_data``` function in decorators.py.  Similar to what's going on in https://github.com/magfest/ubersystem/pull/1853/files